docs/circle_build/README.md

```diff
@@ -0,0 +1,232 @@
+# Overview
+Lets start with the basics, CircleCI is the service we use in order to run our tests and to check the integrity of our code.
+
+As you are probably aware, while pushing any code to our repositories it will initiate a build within CircleCI, this build will execute several things, or in the Circle language steps.
```

Comment:
```suggestion
As you are probably aware, while pushing any code to our repositories it will initiate a build within CircleCI, this build will execute several things, or in Circle jargon, steps.
```